### ISSUE TITLE
If there exists messages when stopDelivery() is called, the node segfault

### ISSUE THREAD

**Issue by hangqi** [hangqi/node-hedwig#1]
If we call stopDelivery() in msgHandler, and there are some more messages pushed from hedwig client, the node will segfault

``` javascript
var client = new hedwig.Hedwig(options. $log4xxConfig);
var cnt = 0;
var msgHandler = function(thisTopic, thisSub, message, consumeCb) {
    cnt += 1;
    console.log('Received message : ' + JSON.stringify(message));
    consumeCb.complete();
    if (cnt == 20) {
        client.stopDelivery(thisTopic, thisSub);
    }   
};

client.sub(topic, subId, hedwig.CREATE_OR_ATTACH, function(error) {
    if (error) {
        console.log('sub failed : ' + error);
    } else {
        console.log('sub succeed !');
        client.startDelivery(topic, subId, msgHandler);
        for (var i = 0; i < 20; i++) {
            client.pub(topic, "msg-" + i, pubCb);
        }   
    }   
});
```

**Comment by hangqi** [comment]
This is because 

HedwigMessageHandler stores a reference of Pesisitent<Function> jsCallback. And it will be wrapped by shared_pointer in hedwig cpp client. When stopDelivery is called, the shared_pointer reference count becomes 0, and thus it will be destroyed, and jsCallback will call Dispose() function. 

But when the HedwigMessageHandler::consume is called inside hedwig thread, it will start eio_nop operation, and then callback in V8 main thread. So when the stopDelivery is called, there may exists some pending eio operations in eio request queue. And when they called in V8 main thread, the jsCallback has been Disposed, thus result in segfault

**Comment by hangqi** [comment]
wrapper the jsCallback as a wrapper, and kept by a shard_pointer

**Comment by hangqi** [comment]
fixed in 48f04bcdceb52e9126bb46978a9a75d522148644
